aries_cloudagent/transport/inbound/session.py

```diff
@@ -169,6 +171,11 @@ async def handle_relay_context(self, payload_enc: Union[str, bytes]):
                     self.profile.context, wallet
                 )
 
+                # Update a responder profile with wallet profile
+                responder: AdminResponder = profile.inject(BaseResponder)
+                responder._profile = profile
```

Comment:
Doesn't this also change the profile for the base wallet? As you're not creating a new responder instance but modifying the old one.

I.e. something like this (not tested):

```suggestion

                base_responder: AdminResponder = profile.inject(BaseResponder)
            
                # Create new responder based on base responder
                responder = AdminResponder(
                    profile,
                    base_responder._send,
                    base_responder._webhook,
                )
                responder._profile = profile
```